cmd/osbuild-worker/jobimpl-depsolve.go

```diff
@@ -42,11 +43,10 @@ func (impl *DepsolveJobImpl) Run(job worker.Job) error {
 	if err != nil {
 		switch err.(type) {
 		case *rpmmd.DNFError:
-			result.ErrorType = worker.DepsolveErrorType
+			result.JobError = clienterrors.WorkerClientError(clienterrors.ErrorDepsolveJob, err.Error())
 		case error:
-			result.ErrorType = worker.OtherErrorType
+			result.JobError = clienterrors.WorkerClientError(clienterrors.ErrorDNFError, err.Error())
```

Comment:
In the general case something went wrong, so a generic ErrorDepsolveJob here would indicate a legit failure that something is up and dnf-json is crashing somewhere it shouldn't.